### PR TITLE
Metrics 2.0 - Deprecate @Counted 

### PR DESCRIPTION
**Pull request by pilhuhn**
DO NOT MERGE - This is meant for the MicroProfile Metrics community to discuss potential changes to the spec and tck.

### CODE REVIEW

**Comment by kenfinnigan** [comment]
When it's ready, should this PR be done against SmallRye?

**Comment by pilhuhn** [comment]
Yes. But I don't have that yet :-)

File: fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/deployment/HitCountedInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return countedCallable(context, context.getMethod());
+    }
+
+    @AroundTimeout
+    private Object countedTimeout(InvocationContext context) throws Exception {
+        return countedCallable(context, context.getMethod());
+    }
+
+    private <E extends Member & AnnotatedElement> Object countedCallable(InvocationContext context, E element) throws Exception {
+        MetricResolver.Of<HitCounted> counted = resolver.hitCounted(bean.getBeanClass(), element);
```
Comment (mkouba) [inline fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/deployment/HitCountedInterceptor.java]: The `@Intercepted Bean<?> bean` may be null - the interceptor should handle this case correctly; see also https://github.com/thorntail/thorntail/blob/master/fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/deployment/CountedInterceptor.java#L73.

**Comment by sberyozkin** [comment]
@pilhuhn Hi, are you planning to continue with this PR ?

**Comment by pilhuhn** [comment]
@sberyozkin Probably not. But I want to first @jmartisk have a look if he can (re) use anything of this for the new "real" implementation in SmallRye.
But then I guess you can close it and he can look at the closed one.

**Comment by jmartisk** [comment]
Thanks; I'll look and perhaps use some inspiration from it. If this should be closed then so be it.

**Comment by sberyozkin** [comment]
@pilhuhn @jmartisk thanks, no problems at all with keeping it open for now. We can close it in a couple of months or so :-) and indeed if needed the closed one will always be here

**Comment by sberyozkin** [comment]
Hi @pilhuhn @jmartisk, it has been nearly 3 months since the last time we discussed the option of keeping this PR open, so if you won't object then I will close it, in the next day or two. It can be reopened at any time...thanks

**Comment by jmartisk** [comment]
Yeah this can be closed, it was pretty much replaced by https://github.com/smallrye/smallrye-metrics/pull/41 in SmallRye

**Comment by sberyozkin** [comment]
Thanks :-)

**Pull request closed without merging**